model-integration/src/main/java/ai/vespa/rankingexpression/importer/tensorflow/TensorFlowImporter.java

```diff
@@ -78,15 +69,18 @@ public ImportedModel importModel(String modelName, String modelDir, SavedModelBu
     private ImportedModel convertToOnnxAndImport(String modelName, String modelDir) {
         Path tempDir = null;
         try {
-            log.info("Converting TensorFlow model '" + modelDir + "' to ONNX...");
             tempDir = Files.createTempDirectory("tf2onnx");
             String convertedPath = tempDir.toString() + File.separatorChar + "converted.onnx";
-            Pair<Integer, String> res = convertToOnnx(modelDir, convertedPath, defaultOnnxOpset);
-            if (res.getFirst() != 0) {
-                throw new IllegalArgumentException("Conversion from TensorFlow to ONNX failed for '" + modelDir + "'. " +
-                        "Reason: " + res.getSecond());
+            for (int opset : onnxOpsetsToTry) {
+                log.info("Converting TensorFlow model '" + modelDir + "' to ONNX with opset " + opset + "...");
+                Pair<Integer, String> res = convertToOnnx(modelDir, convertedPath, opset);
+                if (res.getFirst() == 0) {
+                    log.info("Conversion to ONNX with opset " + opset + " successful.");
+                    return onnxImporter.importModel(modelName, convertedPath);
+                }
+                log.info("Conversion to ONNX with opset " + opset + " failed. Reason: " + res.getSecond());
             }
-            return onnxImporter.importModel(modelName, convertedPath);
+            throw new IllegalArgumentException("Unable to convert TensorFlow model in '" + modelDir + "' to ONNX.");
```

Comment:
Might be useful to include the last reason here?

Reply:
Good suggestion. Fixed.